lib/maths/common/CBayesianOptimisation.cc

```diff
@@ -710,6 +702,28 @@ double CBayesianOptimisation::kernel(const TVector& a, const TVector& x, const T
                                                         .cwiseProduct(x - y));
 }
 
+CBayesianOptimisation::TVector CBayesianOptimisation::kinvf() const {
+    TVector Kinvf;
+    TMatrix K{this->kernel(m_KernelParameters, this->meanErrorVariance())};
+    Kinvf = K.ldlt().solve(this->function());
+    return Kinvf;
+}
+
+CBayesianOptimisation::TVector CBayesianOptimisation::transformTo01(const TVector& x) const {
+    return x - m_MinBoundary.cwiseQuotient(m_MaxBoundary - m_MinBoundary);
+}
+
+double CBayesianOptimisation::dissimilarity(const TVector& x) const {
+    double sum{0.0};
+    double min{0.0};
+    for (const auto& y : m_FunctionMeanValues) {
+        double dxy{las::distance(x, y.first)};
+        sum += dxy;
+        min += std::min(min, -dxy);
```

Comment:
Can you maybe leave a comment on why do you need this `min` term?

Reply:
I added a comment in [2f20ec1](https://github.com/elastic/ml-cpp/pull/2147/commits/2f20ec1ed7e62fe55350232e4874f20b0ee2c476).